thirdparty/mcap/mcap/writer.inl

```diff
@@ -53,7 +56,23 @@ void FileWriter::handleWrite(const std::byte* data, uint64_t size) {
   assert(file_);
   const size_t written = std::fwrite(data, 1, size, file_);
   (void)written;
-  assert(written == size);
+  if (written != size)
+  {
+#ifdef _WIN32
+    std::filesystem::space_info space = std::filesystem::space("C:\\");
+#else
+    std::filesystem::space_info space = std::filesystem::space("/");
+#endif
```

Comment:
I think this should be more precise, a user might have limited the space of a specific folder while still having plenty of space in the system.